hack/local-e2e-common.sh

```diff
@@ -0,0 +1,474 @@
+# This script is sourced by local-e2e-test.sh and local-e2e-pool-test.sh
+
+# Supported cloud platforms (keep in sync with Makefile test-e2e-local target)
+SUPPORTED_CLOUDS="aws/azure/gcp/ibmcloud/vsphere/nutanix/openstack"
+
+# Setup and validate KUBECONFIG
+# If KUBECONFIG environment variable is set, use it; otherwise use default ~/.kube/config
+function setup_kubeconfig() {
+    if [ -n "$KUBECONFIG" ]; then
+        echo "Using KUBECONFIG from environment: $KUBECONFIG"
+        if [ ! -f "$KUBECONFIG" ]; then
+            echo "Error: KUBECONFIG file not found: $KUBECONFIG" >&2
+            exit 1
+        fi
+    else
+        # Auto-detect: use default kubeconfig location if KUBECONFIG is not set
+        DEFAULT_KUBECONFIG="${HOME}/.kube/config"
+        if [ -f "$DEFAULT_KUBECONFIG" ]; then
+            export KUBECONFIG="$DEFAULT_KUBECONFIG"
+            echo "KUBECONFIG not set in environment, using default: $KUBECONFIG"
+        else
+            echo "Error: KUBECONFIG environment variable is not set and default kubeconfig not found at $DEFAULT_KUBECONFIG" >&2
+            echo "Please set KUBECONFIG environment variable:" >&2
+            echo "  export KUBECONFIG=/path/to/your/kubeconfig" >&2
+            exit 1
+        fi
+    fi
+}
+
+# Check cluster connectivity using oc or kubectl
+# Requires KUBECONFIG to be set (should be called after setup_kubeconfig)
+function check_cluster_connectivity() {
+    if [ -z "$KUBECONFIG" ]; then
+        echo "Error: KUBECONFIG is not set. Please call setup_kubeconfig() first." >&2
+        exit 1
+    fi
+    echo "Checking cluster connectivity with kubeconfig: $KUBECONFIG"
+    CLUSTER_ACCESSIBLE=false
+    if command -v oc &> /dev/null; then
+        if KUBECONFIG="$KUBECONFIG" oc cluster-info &> /dev/null; then
+            echo "✓ Cluster is accessible via oc"
+            CLUSTER_ACCESSIBLE=true
+        fi
+    fi
+
+    if [ "$CLUSTER_ACCESSIBLE" = "false" ] && command -v kubectl &> /dev/null; then
+        if KUBECONFIG="$KUBECONFIG" kubectl cluster-info &> /dev/null; then
+            echo "✓ Cluster is accessible via kubectl"
+            CLUSTER_ACCESSIBLE=true
+        fi
+    fi
+
+    if [ "$CLUSTER_ACCESSIBLE" = "false" ]; then
+        if ! command -v oc &> /dev/null && ! command -v kubectl &> /dev/null; then
+            echo "Error: Neither oc nor kubectl found. Please install one of them." >&2
+        else
+            echo "Error: Cannot connect to cluster using KUBECONFIG=$KUBECONFIG" >&2
+            echo "Please verify:" >&2
+            echo "  1. The kubeconfig file is valid" >&2
+            echo "  2. The cluster is accessible" >&2
+            echo "  3. Your credentials are valid" >&2
+        fi
+        exit 1
+    fi
+}
+
+# Setup HIVE_IMAGE: auto-detect, prompt, or use existing value
+# Must export the variable so e2e-common.sh and child processes can access it
+function setup_hive_image() {
+    if [ -z "$HIVE_IMAGE" ]; then
```

Comment:
Here's where I think we could default to `$IMG` since that's what the rest of the Makefile uses.

(While I'm here, I noticed [that Makefile var defaults](https://github.com/openshift/hive/blob/3e0211d5e29b894533e00967f3f950494e30e6f8/Makefile#L49) to something that's almost certainly obsolete for everyone. Can we get rid of that?)